### DISCUSSION TITLE
pytorch for jetson nano orin does not work

### DISCUSSION THREAD

**Discussion by StefanoGarzoni** [roboflow/notebooks#173] [Q&A] [answered]
I bought a jetson nano orin (it seemed like an excellent compromise for not buying a new pc that could handle the power needed to use Yolo) I installed all the useful libraries and also CUDA (version 11.8.89), but when I install torch version: 2.0 .0+nv23.05 and torchvision version : 0.15.1
following the specific instructions for installing on jetson given by nvidia (https://forums.developer.nvidia.com/t/pytorch-for-jetson/72048) and trying to test everything (python3; >>import torch ; >>torch.cuda.is_available() = true) BUT (python3; >>import torchvision =>
gives the following error which I can't fix: 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/__init__.py", line 6, in <module>
    from torchvision import datasets, io, models, ops, transforms, utils
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/datasets/__init__.py", line 1, in <module>
    from ._optical_flow import FlyingChairs, FlyingThings3D, HD1K, KittiFlow, Sintel
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/datasets/_optical_flow.py", line 12, in <module>
    from ..io.image import _read_png_16
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/io/__init__.py", line 8, in <module>
    from ._load_gpu_decoder import _HAS_GPU_VIDEO_DECODER
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/io/_load_gpu_decoder.py", line 1, in <module>
    from ..extension import _load_library
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/extension.py", line 107, in <module>
    _check_cuda_version()
  File "/home/fenixtek/.local/lib/python3.8/site-packages/torchvision-0.15.1-py3.8-linux-aarch64.egg/torchvision/extension.py", line 80, in _check_cuda_version
    raise RuntimeError(
RuntimeError: Detected that PyTorch and torchvision were compiled with different CUDA versions. PyTorch has CUDA Version=11.4 and torchvision has CUDA Version=11.8. Please reinstall the torchvision that matches your PyTorch install.

and I get the same error even if I try to run my py code containing my project based on yolo (the same project you show at the link: https://www.youtube.com/watch?v=Mi9iHFd0_Bo)

if I clearly follow the guide given by pytorch (https://pytorch.org/get-started/locally/) when I then test the result I have (python3; >>import torch ; >>torch.cuda.is_available() = false) and same doing "pip3 install torch" ; "pip3 install torch==2.0.0" ; "pip3 install torchvision" ; "pip3 install torchvision==0.15.1" ; etc etc

so I think that to install pytorch I have to follow the appropriate nvidia guide (as above: https://forums.developer.nvidia.com/t/pytorch-for-jetson/72048), but in any case that error comes out!
I don't know if I need to modify some installation files or download other libraries, I'm desperate!
if I made a trivial mistake, forgive me, I'm still a beginner despite everything.
I hope someone can help me and I thank in advance anyone who can help me solve the problem!

(sorry, I forgot to tell you my Jetpack version : 5.1.1)

**Comment by StefanoGarzoni** [comment, marked as answer]
I solved the problem, in case anyone else needs it, here is the solution:
The problem is that torchvision is compiled with a different version from pythorch [11.8 and 11.4], uninstall torchvision with "pip3 uninstall torchvision" then with "sudo nano /etc/profile" and/or "sudo nano ~/.bashrc" change all PATH containing references to cuda, in cuda-11.4, in this way when following the guide ( https://forums.developer.nvidia.com/t/pytorch-for-jetson/72048 ) torchvision compiles automatically, it will compile on the 11.4 version of cuda which in this case is the version that pytorch is compiled with. Once the installation is finished, test it: $ python3 ; >>import torchvision ; At this point everything works for me!

**Comment by SkalskiP** [comment]
Hi @Stefano0513 👋🏻! I'm sorry I didn't respond earlier. I was super busy with other responsibilities. Nevertheless, it is great to hear that you solved the problem. I'm even more glad that you shared the solution with us. Thank you. 🙏🏻

**Reply by StefanoGarzoni** [reply]
it was a pleasure, thank you for all the times you helped me!
